components/ruby/lib/chef-licensing/restful_client/base.rb

```diff
@@ -176,6 +199,46 @@ def raise_restful_client_conn_error(urls)
 
         raise RestfulClientConnectionError, error_message
       end
+
+      def construct_cache_key(endpoint, params = {})
```

Comment:
Could you move this to a service class? and use that to create cache key something like below, 
```
CacheKey.from(endpoint, params = {})
```

Reply:
if you are building a generic cache key what you say makes sense.. but it is a custom cache key builder specific to License so either you can pass these keys as options or let them read it from some place